taskcluster/rb_taskgraph/gradle.py

```diff
@@ -3,16 +3,52 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
-import json
-import subprocess
-
-from taskgraph.util.memoize import memoize
+# /!\ This implementation differs from what android-components, fenix, and focus have.
+# The main reason is: a-c is much more complex and subject to changes. The 3 projects
+# being in the same repo, they all follow the same model. This means, there's a
+# mechanism to ensure the content of `gradlew printVariants` stays in sync with the
+# .buildconfig.yml files.
+#
+# Here, we're just tracking a handful build_types, so we don't need the same complexity
+# as what was introduced in https://github.com/mozilla-mobile/firefox-android/pull/340.
+#
+# If you want to update this variable, just run `gradlew printVariants` and copy-paste
```

Comment:
Should we have a lint task to validate this stays in sync with the actual gradle config?  I worry we'll forget all about it otherwise, by the time the next relevant change to the gradle config is made, and things will get weird.

Reply:
I understand the concern. If we did port the lint task then we'll have to remember to update the script and the task definition in reference-browser. Therefore, we would be exposed to the same class of bugs: duplication. So the 2 alternatives we have are: 
 1. Remember to maintain the script and the task definition between 2 repos
 2. Remember to maintain a JSON object within a single repo.

Number 2 is the lesser evil to me. That's the main reason why I took this approach. A secondary reason is it also makes the PR easier to review and thus, less risky.

With all that said, I believe a third solution is better in the long run. We should leverage [mozilla-taskgraph](https://github.com/mozilla-releng/mozilla-taskgraph/) and place the script and the task definition in this central place. This way, we're not prone to duplication anymore.